bundles/org.openhab.binding.velbus/src/main/java/org/openhab/binding/velbus/internal/handler/VelbusVMBDALIHandler.java

```diff
@@ -0,0 +1,399 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+
+/**
+ * The {@link VelbusVMBDALIHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Daniel Rosengarten - Initial contribution
+ */
+@NonNullByDefault
+public class VelbusVMBDALIHandler extends VelbusSensorWithAlarmClockHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(
+            Arrays.asList(THING_TYPE_VMBDALI, THING_TYPE_VMBDALI_20));
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @NonNullByDefault({}) VelbusSensorConfig sensorConfig;
+
+    private VelbusColorChannel[] colorChannels;
+    private VelbusVirtualColorChannel[] virtualColorChannels;
+
+    public VelbusVMBDALIHandler(Thing thing) {
+        super(thing, 9);
+
+        colorChannels = new VelbusColorChannel[81];
+        virtualColorChannels = new VelbusVirtualColorChannel[16];
+    }
+
+    @Override
+    public void initialize() {
+        this.sensorConfig = getConfigAs(VelbusSensorConfig.class);
+
+        super.initialize();
+
+        initializeAutomaticRefresh();
+        initializeColorChannel();
+        initializeVirtualLight();
+        initializeChannelStates();
+    }
+
+    private void initializeAutomaticRefresh() {
+        int refreshInterval = this.sensorConfig.refresh;
+
+        if (refreshInterval > 0) {
+            startAutomaticRefresh(refreshInterval);
+        }
+    }
+
+    private void initializeColorChannel() {
+        for (int i = 0; i <= 80; i++) {
+            colorChannels[i] = new VelbusColorChannel();
+        }
+    }
+
+    private void initializeVirtualLight() {
+        String virtualLight;
+
+        for (int i = 1; i <= 16; i++) {
+            if (getConfig().containsKey(VIRTUAL_LIGHT + i)) {
+                virtualLight = getConfig().get(VIRTUAL_LIGHT + i).toString();
+                if (virtualLight.length() > 0) {
+                    try {
+                        virtualColorChannels[i - 1] = new VelbusVirtualColorChannel(virtualLight);
+                    } catch (Exception e) {
+                        logger.error("VMBDALI on address {} : Virtual Light {} has wrong channel format '{}'. {}",
+                                getModuleAddress().getAddress(), i, virtualLight, e.getMessage());
+                    }
+                }
+            }
+            if (virtualColorChannels[i - 1] == null) {
+                virtualColorChannels[i - 1] = new VelbusVirtualColorChannel();
+            }
+        }
+    }
+
+    private void initializeChannelStates() {
+        VelbusBridgeHandler velbusBridgeHandler = getVelbusBridgeHandler();
+        if (velbusBridgeHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
```

Comment:
Pretty sure you do not need to set the bridge to offline, this is done by the framework. Also if it is null, it is rather not initialized. 
Anyway, i suggest to replace this by a warn logger

Reply:
I've reused the same programmation that was written for all the existing modules.

If I'm not wrong, the updateStatus() function do not update the bridge status but of the thing.
Here if there is no bridge defined, we set the thing to OFFLINE. 
It's more the BRIDGE_OFFLINE status wich is not totaly correct, but it's programmed like that for all the modules.

Reply:
I meant the Thing status, not the bridge, i wrote it wrong. Anyway, i also see it is the same with the other modules. Let's leave it as is.